src/main/java/Duke.java

```diff
@@ -1,10 +1,151 @@
+                line = in.nextLine();
+                continue;
+            }
+            // Command is valid, handle the command
+            System.out.println("    ____________________________________________________________");
+            System.out.println("     Got it. I've added this task: ");
+            handleCommand(line);
+            System.out.println("     Now you have " + descriptions.size() + " tasks in the list.");
+            System.out.println("    ____________________________________________________________");
+            line = in.nextLine();
+        }
+        System.out.println("    ____________________________________________________________");
+        System.out.println("     Bye. Hope to see you again soon!");
+        System.out.println("    ____________________________________________________________");
+    }
+
+    private static void handleCommand(String line) {
+        String description;
+        String date;
+        dones.add(false);
+        if (line.startsWith("todo")) {
+            types.add("T");
+            description = line.substring(5);
+            Todo todo = new Todo(description);
+            descriptions.add(description);
+            dates.add("");
+            System.out.println("       " + todo.toString(false));
+        } else if (line.startsWith("deadline")) {
+            types.add("D");
+            int separator = line.indexOf("/by");
+            description = line.substring(9, separator - 1);
+            date = line.substring(separator + 4);
+            Deadline deadline = new Deadline(description, date);
+            descriptions.add(description);
+            dates.add(date);
+            System.out.println("       " + deadline.toString(false));
+        } else if (line.startsWith("event")) {
+            types.add("E");
+            int separator = line.indexOf("/at");
+            description = line.substring(6, separator - 1);
+            date = line.substring(separator + 4);
+            Event event = new Event(description, date);
+            descriptions.add(description);
+            dates.add(date);
+            System.out.println("       " + event.toString(false));
+        } else {
+            types.add("unknown");
+        }
+    }
+
+    private static void showList() {
+        System.out.println("    ____________________________________________________________");
+        System.out.println("     Here are the tasks in your list:");
+        for (int i = 0; i < descriptions.size(); i++) {
+            String description = descriptions.get(i);
+            String date = dates.get(i);
+            String type = types.get(i);
+            boolean isDone = dones.get(i);
+            System.out.printf("     %d.", i + 1);
+            switch (type) {
+                case "T":
+                    Todo todo = new Todo(description);
+                    System.out.println(todo.toString(isDone));
+                    break;
+                case "D":
```

Comment:
No indentation required for case statement

Reply:
Thanks for pointing it out, I will fix it.